pkg/operator/controller/dns/controller.go

```diff
@@ -552,6 +552,7 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			TenantID:       string(creds.Data["azure_tenant_id"]),
 			SubscriptionID: string(creds.Data["azure_subscription_id"]),
 			ARMEndpoint:    platformStatus.Azure.ARMEndpoint,
+			InfraID:        r.infraConfig.Status.InfrastructureName,
```

Comment:
`r.infraConfig` could be nil or out of date.  The caller provides the current infra config status in `infraStatus`.
```suggestion
			InfraID:        infraStatus.InfrastructureName,
```